File: pickerview/src/main/java/org/jaaksi/pickerview/dataset/OptionDataSet.java
package org.jaaksi.pickerview.dataset;

import java.util.List;
import org.jaaksi.pickerview.picker.OptionPicker;

/**
 * Created by fuchaoyang on 2018/2/11.<br/>
 * description：{@link OptionPicker}专用数据集
 */

public interface OptionDataSet extends PickerDataSet {

  /**
   * @return 下一级的数据集
   */
  List<? extends OptionDataSet> getSubs();
}

File: pickerview/src/main/java/org/jaaksi/pickerview/dataset/OptionDataSet.kt
package org.jaaksi.pickerview.dataset

/**
 * Created by fuchaoyang on 2018/2/11.<br></br>
 * description：[OptionPicker]专用数据集
 */
interface OptionDataSet : PickerDataSet {
    /**
     * @return 下一级的数据集
     */
    fun getSubs(): List<OptionDataSet>?
}